tests/integration/component/proxy/proxy_binary_comp.go

```diff
@@ -18,45 +18,95 @@ import (
 	"flag"
 	"fmt"
 	"log"
+	"sync"
 	"time"
 
 	"istio.io/istio/tests/integration/framework"
 )
 
+const (
+	sideCarEndpoint = "http://localhost:9090/echo"
+)
+
 var (
-	envoyBinary = flag.String("envoy_binary", "", "Envoy binary path.")
+	envoyStartScript = flag.String("envoy_start_script", "", "start_envoy script")
+	envoyBinary      = flag.String("envoy_binary", "", "Envoy binary path.")
+	lock             sync.Mutex
 )
 
+// LocalCompConfig contains configs for LocalComponent
+type LocalCompConfig struct {
+	framework.Config
+	LogFile string
+}
+
+// LocalCompStatus contains status for LocalComponent
+type LocalCompStatus struct {
+	framework.Status
+	SideCarEndpoint string
+}
+
 // LocalComponent is a component of local proxy binary in process
 type LocalComponent struct {
 	framework.Component
 	testProcess framework.TestProcess
-	Name        string
-	LogFile     string
+	name        string
+	config      LocalCompConfig
+	status      LocalCompStatus
 }
 
 // NewLocalComponent create a LocalComponent with name and log dir
-func NewLocalComponent(n, logDir string) *LocalComponent {
-	logFile := fmt.Sprintf("%s/%s.log", logDir, n)
-
+func NewLocalComponent(n string, config LocalCompConfig) *LocalComponent {
 	return &LocalComponent{
-		Name:    n,
-		LogFile: logFile,
+		name:   n,
+		config: config,
 	}
 }
 
 // GetName implement the function in component interface
 func (proxyComp *LocalComponent) GetName() string {
-	return proxyComp.Name
+	return proxyComp.name
+}
+
+// GetConfig return the config for outside use
+func (proxyComp *LocalComponent) GetConfig() framework.Config {
+	lock.Lock()
```

Comment:
Perhaps lock is avoidable if you define the contract clearly. Do we expect concurrent get and set?

Reply:
I am thinking the following situations:
1. Some components may have a periodic job keep setting/updating config/status 
2. B is trying to read A's config while it's being set/modified by testcase